Source/Csla.Blazor/ApplicationContextManager.cs

```diff
@@ -43,15 +44,25 @@ public class ApplicationContextManager : IContextManager, IDisposable
     public ApplicationContextManager(AuthenticationStateProvider authenticationStateProvider)
     {
       AuthenticationStateProvider = authenticationStateProvider;
+      CurrentPrincipal = UnauthenticatedPrincipal;
       AuthenticationStateProvider.AuthenticationStateChanged += AuthenticationStateProvider_AuthenticationStateChanged;
-      try
-      {
-        CurrentPrincipal = AuthenticationStateProvider.GetAuthenticationStateAsync().Result.User;
-      }
-      catch
+      InitializeUser();
+    }
+
+    private async void InitializeUser()
```

Comment:
@TheCakeMonster I think this will work better - it avoids the sync over async issue.

Reply:
Hmm, not sure. It's using async void - and I think it's being called synchronously. I think that might risk the same problem, although we're quite far down into the weeds of how async/await works, I know. But I'd say that will still deadlock.

I think attaching a continuation (I think of these like a callback) using ContinueWith on the task that is returned by the GetAuthenticationStateAsync() method avoids the risk. Don't await the task, just tack a continuation/callback onto the end and let that get executed when the task has completed.